portfolio/src/main/webapp/scripts/animations.js

```diff
@@ -33,70 +54,67 @@ function type(id, text, i = 0, isTag = false) {
     isTag = false;
   }
 
-  if (isTag) return type(id, text, i, true); // if tag, add without delay
-  setTimeout(() => type(id, text, i), 100); // if not a tag, add with a delay
+  if (isTag) return type(id, text, startIndex, true); // if tag, add without delay
+  setTimeout(() => type(id, text, startIndex), 100); // if not a tag, add with a delay
 }
 
-/* event handlers */
-
-// Change colors and rotate arrow on hover
-function portfolioBtnHoverOn() {
-  fadeButton('#portfolio-button', '#00bcd4', '#00bcd4', '#ede7f6');
-  gsap.to("#portfolio-arrow", .2, { rotation: "90", ease: Linear.easeNone });
-}
+// --Event handlers--
 
-// Change colors and rotate arrow off hover
-function portfolioBtnHoverOff() {
-  fadeButton('#portfolio-button', 'transparent', '#ede7f6', '#ede7f6');
-  gsap.to("#portfolio-arrow", .2, { rotation: "0", ease: Linear.easeNone });
+/* Change colors and rotate portfolio arrow on hover */
+function portfolioButtonHoverOn() {
+  gsap.to('#portfolio-button', {
+    backgroundColor: PORTFOLIO_HOVER_BLUE, 
+    borderColor: PORTFOLIO_HOVER_BLUE, 
+    color: OFF_WHITE, 
+  })
+  gsap.to("#portfolio-arrow", .2, {rotation: "90", ease: Linear.easeNone});
 }
 
-// Fade a button with the given colors and time
-function fadeButton(id, backgroundColor, borderColor, textColor, easeDuration) {
-  gsap.to(id, { 
-    backgroundColor: backgroundColor, 
-    borderColor: borderColor, 
-    color: textColor, 
-    ease:`power1.out(${easeDuration})` 
+/* Change colors and rotate back portfolio arrow off hover */
+function portfolioButtonHoverOff() {
+  gsap.to('#portfolio-button', {
+    backgroundColor: 'transparent', 
+    borderColor: OFF_WHITE, 
+    color: OFF_WHITE, 
   })
+  gsap.to("#portfolio-arrow", .2, {rotation: "0", ease: Linear.easeNone});
 }
 
-// Change skill dots on/off hover
-function toggleDot(id, years) {
-  const dot = document.getElementById(id);
-  if (dot.classList.contains('flipped')) {
-    dot.classList.remove('flipped');
-    gsap.to("#" + id, 1, { backgroundColor: '#b39ddb' })
-    dot.innerText = id;
+/* Change skill dots on/off hover */
+function toggleSkillDot(skillDot, { language = "", years = -1} = {}) {
```

Comment:
Unless you're planning on adding more different functionality for this, why not have an innerText param instead of this object? I think all you're saving right now is adding ```" years"``` to the end.